cloudinit/helpers.py

```diff
@@ -376,6 +382,12 @@ def _unpickle(self, ci_pkl_version: int) -> None:
             self.run_dir = Paths(
                 path_cfgs=self.cfgs, ds=self.datasource
             ).run_dir
+        if "instance_data" not in self.lookups:
```

Comment:
These lines are a bit confusing at a glance, since in this copy of the code, `instance_data` _is_ in `self.lookups`. This is for loading old pickles that will not contain this data, right? Can we add some comments mentioning that, please?

A separate method `_add_new_fields_to_old_pickles()` to delimit the start and end of upgrade path code might help make this really obvious to future readers, but I'll leave it up to you whether that's necessary.